packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -49,15 +54,33 @@ export default util.createRule<[Options], MessageIds>({
           allowAsThisParameter: {
             type: 'boolean',
           },
+          invalidUnionParameter: {
+            oneOf: [
+              { type: 'boolean' },
+              {
+                type: 'array',
+                items: { type: 'string' },
+                minLength: 1,
+              },
+            ],
+          },
+          allowUnionType: {
+            type: 'boolean',
+          },
         },
         additionalProperties: false,
       },
     ],
   },
   defaultOptions: [
-    { allowInGenericTypeArguments: true, allowAsThisParameter: false },
+    {
+      allowInGenericTypeArguments: true,
+      allowAsThisParameter: false,
+      invalidUnionParameter: true,
+      allowUnionType:true,
+    },
   ],
-  create(context, [{ allowInGenericTypeArguments, allowAsThisParameter }]) {
+  create(context, [{ allowInGenericTypeArguments, allowAsThisParameter, invalidUnionParameter,allowUnionType }]) {
```

Comment:
May I suggest installing the [Prettier](https://prettier.io) extension for your editor? 😄